### DISCUSSION TITLE
Java compatibility : Using the android SDK from java

### DISCUSSION THREAD

**Discussion by joiskash** [google/android-fhir#449] [General]
While implementing `com.google.android.fhir.FhirEngine`  in a java class, every function has an extra argument `Continuation<? super Bundle> continuation` that cannot be null. This maybe due to the `suspend` keyword. More details are available in this [link](https://jorgecastillo.dev/digging-into-kotlin-continuations).

**Comment by yigit** [comment]
Hi,

We are aware that the library is not very usable outside kotlin coroutines right now.

We plan to tackle it in the future by writing Java / RX wrappers (similar to AndroidX) once the library API becomes more stable. 

Is this blocking you? If so, we can help unblock you by creating a wrapper that handles your use case instead of a generic one for now.

**Reply by joiskash** [reply]
Ok thats alright. I have currently written a kotlin wrapper and it seems to be working. Im not sure if thats the right way though. The wrapper gets the application context as a parameter .
```
val coroutineScope: CoroutineScope
    get() {
      return MainScope() + CoroutineName(context.javaClass.simpleName)
    }

  fun updateFromJava( resourceToUpdate: Resource){
    coroutineScope.launch {
      fhirEngine.update(resourceToUpdate)
    }
  }

  fun searchFromJava(search: Search): List<Patient> {
    var searchResults: List<Patient> = listOf()

    runBlocking {
      searchResults =
        fhirEngine.search(
          search
        )
    }
    return searchResults
  }
``` 

Do let me know if there is a better way of doing this. Thanks!
